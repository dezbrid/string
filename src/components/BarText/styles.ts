import {StyleSheet} from 'react-native';
import {extraLightGray, white} from '@constants/colors';
import {GENERAL_BOX_SHADOW} from '@constants/commonStyles';

const CLEAR_ICON_SIZE = 18;

export default StyleSheet.create({
  container: {
    margin: 20,
  },
  barText: {
    alignItems: 'center',
    backgroundColor: white,
    height: 45,
    borderRadius: 20,
    flexDirection: 'row',
    paddingRight: 15,
    paddingLeft: 10,
  },
  inputStyle: {
    flex: 1,
    color: 'black',
  },
  clearIcon: {
    width: CLEAR_ICON_SIZE,
    height: CLEAR_ICON_SIZE,
  },
  barTextShadow: {
    ...GENERAL_BOX_SHADOW,
    borderWidth: 1,
    borderColor: extraLightGray,
  },
});
