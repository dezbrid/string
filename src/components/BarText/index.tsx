import React from 'react';
import {View, Image, TextInput, TouchableOpacity} from 'react-native';
import {mediumGray} from '@constants/colors';

import clearIcon from './assets/ic_close.png';
import styles from './styles';

interface Props {
  text: string;
  handleChangeText: (t: string) => void;
}

function BarText({text, handleChangeText}: Props) {
  return (
    <View style={styles.container}>
      <View style={[styles.barText, styles.barTextShadow]}>
        <TextInput
          style={styles.inputStyle}
          placeholder={'ingrese el texto aquÃ­'}
          placeholderTextColor={mediumGray}
          onChangeText={handleChangeText}
          value={text}
          testID="input"
        />
        {text.length > 0 && (
          <TouchableOpacity
            testID={'clearIcon'}
            onPress={() => handleChangeText('')}>
            <Image
              source={clearIcon}
              resizeMode="contain"
              style={styles.clearIcon}
            />
          </TouchableOpacity>
        )}
      </View>
    </View>
  );
}
export default BarText;
