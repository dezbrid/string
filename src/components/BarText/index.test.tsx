import 'react-native';
import React from 'react';
import {render, fireEvent, waitFor} from '@testing-library/react-native';

import BarText from './index';

describe('<BarText />', () => {
  const handleChangeText = jest.fn();
  test('<BarText /> snapshot', () => {
    const tree = render(
      <BarText text={''} handleChangeText={handleChangeText} />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('should can see Test', () => {
    const {findByText} = render(
      <BarText text={'Test'} handleChangeText={handleChangeText} />,
    );
    expect(findByText('Test')).toBeTruthy();
  });
  test('should used button', () => {
    const {getByTestId} = render(
      <BarText text={'Test'} handleChangeText={handleChangeText} />,
    );
    const button = getByTestId('clearIcon');
    fireEvent.press(button);
    waitFor(() => {
      expect(handleChangeText).toHaveBeenCalled();
    });
  });
});
