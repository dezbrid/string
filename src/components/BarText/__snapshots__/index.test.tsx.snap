// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BarText /> <BarText /> snapshot 1`] = `
<View
  style={
    Object {
      "margin": 20,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 20,
          "flexDirection": "row",
          "height": 45,
          "paddingLeft": 10,
          "paddingRight": 15,
        },
        Object {
          "borderColor": "#F1F1F1",
          "borderWidth": 1,
          "shadowColor": "#939393",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
        },
      ]
    }
  >
    <TextInput
      onChangeText={[MockFunction]}
      placeholder="ingrese el texto aquÃ­"
      placeholderTextColor="#939393"
      style={
        Object {
          "color": "black",
          "flex": 1,
        }
      }
      testID="input"
      value=""
    />
  </View>
</View>
`;
